File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/RegisterZoneUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.RegisterZoneCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.ZoneResult;
import reactor.core.publisher.Mono;

public interface RegisterZoneUseCase {

    Mono<ZoneResult> register(RegisterZoneCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/RegisterZoneUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.RegisterZoneCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.ZoneResult
import reactor.core.publisher.Mono

interface RegisterZoneUseCase {
    fun register(command: RegisterZoneCommand): Mono<ZoneResult>
}
